app/services/bookings/gitis/school_experience.rb

```diff
@@ -0,0 +1,92 @@
+module Bookings
+  module Gitis
+    # Creates a new 'Candidate School Experience' entity in Gitis. We should create a new
+    # entity when the status of the school experience changes.
+    class SchoolExperience
+      include ActiveModel::Validations
+
+      private_class_method :new
+
+      class << self
+        def from_placement_request(request, status)
+          new(
+            date: request.placement_date&.date,
+            urn: request.school.urn,
+            school_name: request.school.name,
+            teaching_subject_name: request.subject_first_choice,
+            duration: request.placement_date&.duration,
+            status: status
+          )
+        end
+
+        def from_booking(booking, status)
+          new(
+            date: booking.date,
+            urn: booking.bookings_school.urn,
+            school_name: booking.bookings_school.name,
+            teaching_subject_name: booking.bookings_subject.name,
+            duration: booking.duration,
+            status: status
+          )
+        end
+
+        def from_cancellation(cancellation, status)
+          new(
+            date: (cancellation.booking || cancellation.placement_request.placement_date)&.date,
+            urn: cancellation.school_urn,
+            school_name: cancellation.school_name,
+            teaching_subject_name: cancellation.booking.bookings_subject.name,
+            duration: cancellation.booking.duration,
+            status: status
+          )
+        end
+      end
+
+      def write_to_gitis_contact(contact_id)
+        api = GetIntoTeachingApiClient::SchoolsExperienceApi.new
+        api.add_school_experience(contact_id, gitis_school_experience)
+      end
+
+    private
+
+      GITIS_STATUS = {
```

Comment:
Can we shift lines 52-67 up to the top of the class? its a bit misleading having them under `private` as they aren't going to be